pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,210 @@
+"""
+Internal function to load GMT remote datasets.
+"""
+from typing import Dict, NamedTuple
+
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import kwargs_to_strings
+from pygmt.io import load_dataarray
+from pygmt.src import grdcut, which
+
+
+class Resolution(NamedTuple):
+    """
+    The available grid registrations for a given resolution and whether it is a
+    tiled grid.
+
+    Parameters
+    ----------
+    registrations : list
+        A list of the accepted registrations for a given resolution.
+        Can be either "pixel" or "gridline".
+
+    tiled : bool
+        States if the given resolution is tiled, which requires an
+        argument for ``region``."
+    """
+
+    registrations: list[str]
+    tiled: bool
+
+
+class Dataset(NamedTuple):
+    """
+    Standard information about a dataset and grid metadata.
+
+    Parameters
+    ----------
+    title : str
+        The title of the dataset, used in error messages.
+
+    name : str
+        The name assigned as an attribute to the data array.
+
+    long_name : str
+        The long name assigned as an attribute to the data array.
+
+    units : str
+        The units of the values in the data array.
+
+    resolutions : dict
+        Dictionary of with given resolution as keys and the values use
+        the Resolution class.
+
+    vertical_datum : str
+        The datum used on the vertical axis.
+
+    horizontal_datum : str
+        The datum used on the horizontal axis.
+    """
+
+    title: str
+    name: str
+    long_name: str
+    units: str
+    resolutions: Dict[str, Resolution]
+    vertical_datum: str
+    horizontal_datum: str
+
+
+datasets = {
+    "earth_relief": Dataset(
+        title="Earth relief",
+        name="elevation",
+        long_name="elevation relative to the geoid",
```

Comment:
Just in case that we will have "Mars elevation" in the future:
```suggestion
        long_name="Earth elevation relative to the geoid",
```